### PR TITLE
Add build-image command for Go SDK

### PR DESCRIPTION
**Pull request by bergundy**
- Use temporalite from temporalio org to get non-CGo version of SQLite driver

Checklist:
- [x] settle on label schema and [lint labels](https://github.com/hadolint/hadolint#linting-labels)
- [x] run against cloud namespace
- [x] run in parallel
- [x] set up workflow manual trigger

### CODE REVIEW

File: cmd/build_image.go [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+
+	return i.dockerBuild(ctx, buildConfig{
+		tags:      tags,
+		labels:    map[string]string{"SDK_REPO_URL": i.config.RepoURL, "SDK_REPO_REF": repoRef},
+		buildArgs: map[string]string{"SDK_VERSION": repoDir, "REPO_DIR_OR_PLACEHOLDER": repoDir},
+	})
+}
+
+func (i *ImageBuilder) buildFromVersion(ctx context.Context) error {
+	version := semver.Canonical(i.config.Version)
```
Comment (cretz) [inline cmd/build_image.go:162]: I don't think you should validate this and I don't think you should use the Go library which requires a `v` prefix. Can't we just use the version as the tag directly? We intentionally don't validate version in other SDK features commands.

File: cmd/build_image.go [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+	}
+	for k, v := range config.labels {
+		args = append(args, "--label", fmt.Sprintf("%s=%s", k, v))
+	}
+	for k, v := range config.buildArgs {
+		args = append(args, "--build-arg", fmt.Sprintf("%s=%s", k, v))
+	}
+	args = append(args, i.rootDir)
+
+	i.log.Info("building docker image", tag.NewAnyTag("args", args))
```
Comment (cretz) [inline cmd/build_image.go:229]: Should capitalize logs IMO

**Review by cretz — APPROVED** [review]
Looks good.

Should be noted that you are still using the internet to download Go dependencies when run. So the need for a version-specific docker build makes less sense for Go than it does for languages where there's actually a pre-build step.

File: cmd/build_image.go (hunk 1 above)
Comment (bergundy) [reply cmd/build_image.go:162]: I need this for generating the different semver tags (1, 1.16, 1.16.0), All of our SDKs support semver except for Python for now (granted the `v` prefix is go specific).
I will probably end up changing this a bit when I implement building containers for another language.

**Comment by bergundy** [comment]
> Looks good.
> 
> Should be noted that you are still using the internet to download Go dependencies when run. So the need for a version-specific docker build makes less sense for Go than it does for languages where there's actually a pre-build step.

Hmmm.. what makes you say that? I can run the container with no internet access (turned wifi off on my laptop).
Tests also start immediately and nothing build-related is logged.

File: cmd/build_image.go (hunk 1 above)
Comment (cretz) [reply cmd/build_image.go:163]: Ah, Docker tags? :+1:

**Comment by cretz** [comment]
> I can run the container with no internet access (turned wifi off on my laptop).

Hrmm, Unless you are copying over the go mod cache, it should be reaching out to the internet to download dependencies. But I may be confused on what exactly is being downloaded when.

File: .github/workflows/docker-images.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      TEMPORAL_CLOUD_ADDRESS: sdk-ci.a2dd6.tmprl.cloud:7233
+      TEMPORAL_CLOUD_NAMESPACE: sdk-ci.a2dd6
+      REPO_URL: ${{ github.event.pull_request.head.repo.html_url }}
+    steps:
+      - name: Print build information
+        run: 'echo head_ref: ${{ github.head_ref }}, ref: ${{ github.ref }}'
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '^1.17'
```
Comment (cretz) [inline .github/workflows/docker-images.yaml:45]: Should bump up to at least 1.18 (1.17 is EOL, I need to fix in other places)

File: cmd/build_image.go (hunk 1 above)
Comment (bergundy) [reply cmd/build_image.go:167]: Yup

**Review by cretz — APPROVED** [review]

File: .github/workflows/docker-images.yaml (hunk 3 above)
Comment (bergundy) [reply .github/workflows/docker-images.yaml:45]: I'll fix in one of the next PRs.

**Comment by bergundy** [comment]
Python tests are still broken, force merged.

**Pull request merged**